### PR TITLE
1445: Field cleanup

### PR DESCRIPTION
**Pull request by xendk**
This cleans up ting object/collection field handling.

hook_ding_entity_fields() is removed and the existing fields are unlocked. Proper virtual_field definitions has been added to allow admins/site builders to add the virtual fields from the "Manage fields" backend.

ding_ting_frontend has been re-exported with unlocked fields.

Field titles in ting_material_details has been streamlined to be consistent.

### CODE REVIEW

**Comment by xendk** [comment]
I'll fix a few of the Scrutinizer errors, but I can't just rename DingEntity::ding_entity_id to DingEntity::dingEntityId without servere consequences.

**Comment by xendk** [comment]
Fixed. The last two code style issues are ding_entity_id.

File: modules/ding_availability/ding_availability.install [hunk 1]
```diff
@@ -36,3 +17,12 @@ function ding_availability_update_7000() {
     ->condition('module', 'ding_availability')
     ->execute();
 }
+
+/**
+ * Unlock fields, see ding_entity_unlock_fields().
+ */
```
Comment (cableman) [inline modules/ding_availability/ding_availability.install]: Should the see have been a @see on it's one line for those who generate the doc-blocks!

File: modules/ding_entity/ding_entity.install [hunk 2]
```diff
@@ -35,4 +16,11 @@ function ding_entity_update_7000() {
     ))
     ->condition('module', 'ding_entity')
     ->execute();
-}
\ No newline at end of file
+}
+
+/**
+ * Unlock fields, see ding_entity_unlock_fields().
+ */
```
Comment (cableman) [inline modules/ding_entity/ding_entity.install]: See ding_availability.install for comment about doc-block.

File: modules/ding_periodical/ding_periodical.install [hunk 3]
```diff
@@ -36,4 +16,11 @@ function ding_periodical_update_7000() {
     ))
     ->condition('module', 'ding_periodical')
     ->execute();
-}
\ No newline at end of file
+}
+
+/**
+ * Unlock fields, see ding_entity_unlock_fields().
```
Comment (cableman) [inline modules/ding_periodical/ding_periodical.install]: See ding_availability.install for comment about doc-block.

File: modules/ding_ting_frontend/ding_ting_frontend.info [hunk 4]
```diff
@@ -86,3 +86,4 @@ features[variable][] = page_manager_search_disabled_node
 features[variable][] = page_manager_user_view_disabled
 features[variable][] = pm_existing_pages_disabled_fulltext
 features[variable][] = pm_existing_pages_disabled_infomedia
+mtime = 1448043917
```
Comment (cableman) [inline modules/ding_ting_frontend/ding_ting_frontend.info]: Please remove the mtime line.

File: modules/ting/ting.install [hunk 5]
```diff
@@ -354,3 +243,20 @@ function ting_update_7009() {
   ));
   db_add_unique_key('ting_object', 'vid', array('vid'));
 }
+
+/**
+ * Unlock fields, see ding_entity_unlock_fields().
```
Comment (cableman) [inline modules/ting/ting.install]: See ding_availability.install for comment about doc-block.

File: modules/ting_infomedia/ting_infomedia.install [hunk 6]
```diff
@@ -65,13 +46,8 @@ function ting_infomedia_update_7000() {
 }
 
 /**
- * Unlock ding_entity_field to allow deletion of field
+ * Unlock fields, see ding_entity_unlock_fields().
  */
 function ting_infomedia_update_7001() {
-  db_update('field_config')
```
Comment (cableman) [inline modules/ting_infomedia/ting_infomedia.install]: Update an update function. But if the sites have execute this update. They won't get this updated update?

File: modules/ting_reference/ting_reference.install [hunk 7]
```diff
@@ -35,3 +16,10 @@ function ting_reference_update_7000() {
     ->condition('module', 'ting_reference')
     ->execute();
 }
+
+/**
+ * Unlock fields, see ding_entity_unlock_fields().
```
Comment (cableman) [inline modules/ting_reference/ting_reference.install]: See ding_availability.install for comment about doc-block.

File: modules/ting_relation/ting_relation.install [hunk 8]
```diff
@@ -35,3 +16,10 @@ function ting_relation_update_7000() {
     ->condition('module', 'ting_relation')
     ->execute();
 }
+
+/**
+ * Unlock fields, see ding_entity_unlock_fields().
```
Comment (cableman) [inline modules/ting_relation/ting_relation.install]: See ding_availability.install for comment about doc-block.

File: modules/ting_sfx/ting_sfx.install [hunk 9]
```diff
@@ -59,3 +59,12 @@ function ting_sfx_update_7000() {
     ->condition('module', 'ting_sfx')
     ->execute();
 }
+
+/**
+ * Unlock fields, see ding_entity_unlock_fields().
```
Comment (cableman) [inline modules/ting_sfx/ting_sfx.install]: See ding_availability.install for comment about doc-block.

**Comment by cableman** [comment]
Just some minor comments else the code looks good :-)

**Comment by cableman** [comment]
Also please add the missing issue number to the last commit message.

File: modules/ting_infomedia/ting_infomedia.install (hunk 6 above)
Comment (xendk) [inline modules/ting_infomedia/ting_infomedia.install]: The new version does the same as the old, but in the same way as all the other new update functions. The functionality haven't changed, the implementation just adjusted for consistency.

**Comment by xendk** [comment]
Fixed.

File: modules/ting_infomedia/ting_infomedia.install (hunk 6 above)
Comment (cableman) [inline modules/ting_infomedia/ting_infomedia.install]: Okay, just wanted to be sure :-)

**Comment by cableman** [comment]
Reviewed and awaiting merge

**Pull request merged**